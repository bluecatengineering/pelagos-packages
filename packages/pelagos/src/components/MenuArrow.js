import PropTypes from 'prop-types';

import './MenuArrow.less';

const MenuArrow = ({className}) => (
	<svg className={`MenuArrow${className ? ` ${className}` : ''}`} viewBox="0 0 7 5">
		<path d="m 0 1.5 l 1.1 -1.1 l 2.4 2.4 l 2.4 -2.4 l 1.1 1.1 l -3.5 3.5 z" />
	</svg>
);

MenuArrow.propTypes = {
	/** The component class name(s). */
	className: PropTypes.string,
};

export default MenuArrow;
