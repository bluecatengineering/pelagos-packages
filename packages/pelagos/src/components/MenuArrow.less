@layer pelagos {
	.MenuArrow {
		width: 7px;
		height: 5px;
		fill: currentcolor;
		pointer-events: none;
	}
}
