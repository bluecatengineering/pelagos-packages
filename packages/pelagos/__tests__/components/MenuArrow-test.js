import {shallow} from 'enzyme';

import MenuArrow from '../../src/components/MenuArrow';

jest.unmock('../../src/components/MenuArrow');

describe('MenuArrow', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<MenuArrow />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});

		it('renders expected elements when optional properties are set', () => {
			const wrapper = shallow(<MenuArrow className="TestClass" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
