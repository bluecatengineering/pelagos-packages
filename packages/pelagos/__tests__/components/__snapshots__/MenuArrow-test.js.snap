// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuArrow rendering renders expected elements 1`] = `
<svg
  className="MenuArrow"
  viewBox="0 0 7 5"
>
  <path
    d="m 0 1.5 l 1.1 -1.1 l 2.4 2.4 l 2.4 -2.4 l 1.1 1.1 l -3.5 3.5 z"
  />
</svg>
`;

exports[`MenuArrow rendering renders expected elements when optional properties are set 1`] = `
<svg
  className="MenuArrow TestClass"
  viewBox="0 0 7 5"
>
  <path
    d="m 0 1.5 l 1.1 -1.1 l 2.4 2.4 l 2.4 -2.4 l 1.1 1.1 l -3.5 3.5 z"
  />
</svg>
`;
